tests/unit/test_hardware.py

```diff
@@ -67,6 +67,26 @@ def test_constraint_name_pattern(value: str, expected: tuple[Any, Any]) -> None:
     assert match.groups() == expected
 
 
+_size_constraint_pattern_input = [
+    ({'name': 'num_with_default', 'raw_value': '10', 'default_unit': 'GiB'},
+     'num_with_default: == 10 gibibyte'),
+    ({'name': 'num_without_default', 'raw_value': '1024'}, 'num_without_default: == 1024 byte'),
+    ({'name': 'num_with_unit', 'raw_value': '10 GiB', 'default_unit': 'GiB'},
```

Comment:
I would suggest using a different `default_unit` to make it clear that the unit from `raw_value` is being used.
```suggestion
    ({'name': 'num_with_unit', 'raw_value': '10 GiB', 'default_unit': 'MiB'},
```